### DISCUSSION TITLE
cli: how to save a db in duckdb format, after working in memory

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#8508] [Q&A] [answered]
I open duckdb cli without specifying db name. Therefore, I'm working on a memory db.
Then I create some tables in this memory db.

Is there a way to save all in mydb.duckdb file, before closing this memory duckdb session?

Thank you

**Comment by StLeoX** [comment]
https://duckdb.org/docs/sql/statements/copy#copy-to

**Reply by aborruso** [reply]
> [duckdb.org/docs/sql/statements/copy#copy-to](https://duckdb.org/docs/sql/statements/copy#copy-to)

I have a multi tables memory db. I want to create a duckdb file. I don't think it can be done using "copy". Am I wrong?

**Reply by Tishj** [reply]
https://duckdb.org/docs/sql/statements/export

This might be what you're looking for

**Reply by aborruso** [reply]
> duckdb.org/docs/sql/statements/export
>
> This might be what you're looking for

@Tishj I have tried it before asking it here, and I'm able to export only in CSV and parquet.
Are you able to export in duckdb format? If yes, could you share a sample command?

Thank you

**Reply by Tishj** [reply]
No you can't export directly to duckdb format.
What you can do is EXPORT DATABASE (preferably format parquet because it's more efficient), start a persistent database (`./duckdb my_db.duckdb`) and then IMPORT DATABASE

You could also just directly start a persistent database and skip these steps

**Reply by aborruso** [reply]
Ok, if I open duckdb cli without specifying db name, there is currently no way to export a duckdb file in one step.

Thank you @Tishj

**Comment by l1t1** [comment]
. dump to a sql

**Reply by aborruso** [reply]
Then there is no way to create a duckdb file in one step?

**Comment by l1t1** [comment]
if you use python, you may open two db, of which one is memory db,  the other is db on disk, then write a script to fill one with the other.

**Comment by aborruso** [comment]
If I open duckdb cli without specifying db name, there is currently no way to export the memory db as duckdb file in one step.

**Comment by l1t1** [comment]
btw sqlite can do it
.save ?OPTIONS? FILE     Write database to FILE (an alias for .backup ...)

**Comment by l1t1** [comment]
sqlite CLI is supported in python 3.12
d:\python312>python -m sqlite3
sqlite3 shell, running on SQLite version 3.39.4
Connected to a transient in-memory database

Each command will be run using execute() on the cursor.
Type ".help" for more information; type ".quit" or CTRL-D to quit.
sqlite>

**Comment by gonsalu** [comment, marked as answer]
Yes, this is possible to do now: https://duckdb.org/docs/stable/guides/snippets/copy_in-memory_database_to_file.html

```
ATTACH 'my_database.db';
COPY FROM DATABASE memory TO my_database;
DETACH my_database;
```

**Reply by aborruso** [reply]
thank you
